app/src/main/java/org/fossasia/openevent/adapters/ScheduleSessionsListAdapter.java

```diff
@@ -0,0 +1,118 @@
+    private List<Session> originalList, displayedList;
+
+    @SuppressWarnings("all")
+    Filter filter = new Filter() {
+        @Override
+        protected FilterResults performFiltering(CharSequence constraint) {
+            DbSingleton instance = DbSingleton.getInstance();
+            // TODO: Use a query to do this, iterating over an entire set is pretty bad
+            final ArrayList<Session> filteredSessionList = new ArrayList<>();
+            String query = constraint.toString().toLowerCase();
+            for (Session session : displayedList) {
+                final String text = session.getTitle().toLowerCase();
+                if (text.contains(query)) {
+                    filteredSessionList.add(session);
+                }
+            }
+            FilterResults filterResults = new FilterResults();
+            filterResults.values = filteredSessionList;
+            filterResults.count = filteredSessionList.size();
+            Timber.d("Filtering done total results %d", filterResults.count);
+            return filterResults;
+        }
+
+        @Override
+        protected void publishResults(CharSequence constraint, FilterResults results) {
+            animateTo((List<Session>) results.values);
+        }
+    };
+
+    public ScheduleSessionsListAdapter(List<Session> sessions) {
+        super(sessions);
+        this.originalList = new ArrayList<>(sessions);
+        this.displayedList = sessions;
+    }
+
+    @Override
+    public Filter getFilter() {
+        return filter;
+    }
+
+    @Override
+    public Viewholder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+        View view = layoutInflater.inflate(R.layout.item_session_with_bookmark, parent, false);
+        return new Viewholder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(Viewholder holder, int position) {
+        Session current = getItem(position);
+        SimpleDateFormat formatSDF = new SimpleDateFormat("hh:mm aa", Locale.getDefault());
+        String startTime = formatSDF.format(ISO8601Date.getDateObject(current.getStartTime())),
+                endTime = formatSDF.format(ISO8601Date.getDateObject(current.getEndTime()));
+
+        TypedArray ta = holder.itemView.getContext().getResources().obtainTypedArray(R.array.tracks_colors);
+        holder.trackIndicator.setBackgroundColor(ta.getColor(current.getTrack() - 1, Color.GRAY));
+        ta.recycle();
+
+        holder.sessionTitle.setText(current.getTitle());
+        holder.sessionTime.setText(String.format("%s - %s", startTime, endTime));
+        holder.sessionLocation.setText(DbSingleton.getInstance().getMicrolocationById(current.getMicrolocations()).getName());
+    }
+
+    class Viewholder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        View trackIndicator;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/triplez/open-event-android/file/2406802064/issues/source?bid=3233850&fileBranchId=3233911#l93)